Added WP Admin page load tests

MrJnrman opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR adds the `runAdminWcPageTests` test that clicks links in the sidebar for various top-level menu items and verifies that pages load.

This PR also implements a workaround where the test script sets the viewport size in its test block using the code below.
```
page.setViewport( {
	width: 1280,
	height: 800,
} ),
```

This is done because `plugins/woocommerce/tests/e2e/config/jest.setup.js`  sets the viewport size using `await setBrowserViewport( 'large' );`. Jest defines the dimensions of *large* as  `960x700` which is too small a viewport to allow the side-menu to be visible. This workaround can be removed when #31305 is closed.

Closes #29319.

### How to test the changes in this Pull Request:
1. Run the following commands.
```
$ cd plugins/woocommerce
$ pnpm install
$ pnpx wc-e2e docker:up
$ pnpx wc-e2e test:e2e
```

2. All tests should pass.
### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Added a `specs/data` folder to store page element data.
> Added tests to verify that different top-level menu and their associated sub-menus load successfully.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

MrJnrman committed:
Added WooCommerce menu page load test script

MrJnrman committed:
Added function to open WC Admin Page

MrJnrman committed:
Updated WooCommerce menu page load script

MrJnrman committed:
Added runWcPageLoadTest & runAdminPageLoadTests to exports

MrJnrman committed:
Added page load tests to the plugin e2e directory

MrJnrman committed:
Renamed & updated to test multiple menus

This script takes a menus array which contains selectors for top-level menu items and their associated sub-menus and verifies that they all load successfully.

MrJnrman committed:
Removed unused function

MrJnrman committed:
Updated function name to match its export

rrennick commented on packages/js/e2e-core-tests/specs/index.js:
```diff
@@ -47,6 +47,7 @@ const runOrderSearchingTest = require( './merchant/wp-admin-order-searching.test
 const runAnalyticsPageLoadsTest = require( './merchant/wp-admin-analytics-page-loads.test' );
 const runImportProductsTest = require( './merchant/wp-admin-product-import-csv.test' );
 const runInitiateWccomConnectionTest = require( './merchant/wp-admin-extensions-connect-wccom.test' );
+const runPageLoadTest = require( './merchant/wp-admin-page-loads.test.js' );
```
This could be aliased to `runAdminPageLoadTests`

rrennick commented on packages/js/e2e-core-tests/specs/index.js:
```diff
@@ -166,4 +171,6 @@ module.exports = {
 	runOrderEmailReceivingTest,
 	runInitiateWccomConnectionTest,
 	runTelemetryAPITest,
+	runPageLoadTest,
```
This export is effectively a duplicate.

rrennick commented on packages/js/e2e-core-tests/specs/merchant/wp-admin-page-loads.test.js:
```diff
@@ -0,0 +1,131 @@
+	describe.each( menus )(
+		' %s > Opening Top Level Pages',
+		( menuTitle, menuElement, subMenus ) => {
+			beforeAll( async () => {
```
We should log out the merchant in `afterAll()`

rrennick commented on packages/js/e2e-core-tests/specs/merchant/wp-admin-page-loads.test.js:
```diff
@@ -0,0 +1,131 @@
+					await Promise.all( [
+						page.click( menuElement ),
+						page.waitForNavigation( { waitUntil: 'networkidle0' } ),
+						page.setViewport( {
```
Can you check to see if this will work in the `beforeAll()`

MrJnrman replied:
The viewport is now being set in the `beforeAll` block, but now, page load tests are failing because the side menu isn't expanded. My assumption is that the viewport is being resized in between individual test steps.

MrJnrman committed:
Added elements.js to store element data

MrJnrman committed:
Aliased runPageLoadTest to runAdminPageLoadTests

MrJnrman committed:
Moved setViewport and logged out after each test

Changes:
- Moved setViewport to the beforeAll block
- Added a afterAll block to log out after each set of tests
- References the MENUS array from specs/data/elements.js

MrJnrman commented on packages/js/e2e-core-tests/specs/data/elements.js:
```diff
@@ -0,0 +1,85 @@
+				'#toplevel_page_woocommerce > ul > li:nth-child(8)',
+				'System status',
+			],
+			// [ 'Extensions', '#toplevel_page_woocommerce > ul > li:nth-child(9)', 'Extensions' ],
```
@TODO

Uncomment this line when the issue outlined [here](https://github.com/woocommerce/woocommerce/issues/31304) is fixed.

MrJnrman committed:
Updated CHANGELOG

MrJnrman commented on packages/js/e2e-core-tests/specs/merchant/wp-admin-page-loads.test.js:
```diff
@@ -0,0 +1,53 @@
+		( menuTitle, menuElement, subMenus ) => {
+			beforeAll( async () => {
+				await merchant.login();
+				await page.setViewport( {
```
@Todo

Remove setting the viewport when this [issue](https://github.com/woocommerce/woocommerce/issues/31305) is fixed.

rrennick commented:
@MrJnrman In the review yesterday I didn't think to mention that the package readme files have lists of available functions that you will want to update:

- https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/e2e-core-tests/README.md#merchant
- https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/e2e-utils/README.md#merchant-merchant

MrJnrman committed:
Added runAdminPageLoadTests to runMerchantTests

MrJnrman commented:
Thanks for the reminder. I've gone ahead and updated the readme files accordingly.

MrJnrman committed:
Updated css selectors

MrJnrman committed:
Moved seViewport to the test block

rrennick commented on packages/js/e2e-core-tests/specs/merchant/wp-admin-page-loads.test.js:
```diff
@@ -0,0 +1,53 @@
+				}
+			);
+
+			afterAll( async () => {
```
In our other tests we have been putting the `afterAll()` code block after the `beforeAll()`.

rrennick reviewed (dismissed):
@MrJnrman Nice work on this. The only nitpick is the `afterAll()`. I am pre-approving the PR and you can merge it once you've moved that block of code.

MrJnrman committed:
Moved afterAll block to after beforeAll
